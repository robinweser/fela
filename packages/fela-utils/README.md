# fela-utils

<a href="https://bundlephobia.com/result?p=fela-utils@latest"><img alt="Bundlephobia" src="https://img.shields.io/bundlephobia/minzip/fela-utils.svg"></a>

fela-utils is a private utility package for Fela and should only be used interally.


## License
Fela is licensed under the [MIT License](http://opensource.org/licenses/MIT).<br>
Documentation is licensed under [Creative Commons License](http://creativecommons.org/licenses/by/4.0/).<br>
Created with â™¥ by [@robinweser](http://weser.io) and all the great contributors.
