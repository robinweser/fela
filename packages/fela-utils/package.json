{
  "name": "fela-utils",
  "version": "11.5.0-rc.0",
  "description": "Internal utilities for Fela",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "sideEffects": false,
  "files": [
    "LICENSE",
    "README.md",
    "lib/**",
    "es/**"
  ],
  "repository": "https://github.com/robinweser/fela/",
  "keywords": [
    "fela",
    "fela-utils",
    "cssinjs"
  ],
  "author": "Robin Weser",
  "license": "MIT",
  "dependencies": {
    "css-in-js-utils": "^3.0.0",
    "fast-loops": "^1.0.0"
  },
  "gitHead": "4cc4686d9e16877251ac5c73e161076ec7c113a1"
}
