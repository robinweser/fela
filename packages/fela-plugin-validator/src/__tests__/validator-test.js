import { RULE_TYPE, KEYFRAME_TYPE } from 'fela-utils'
import validator from '../index'

describe('Validator plugin', () => {
  const consoleSpy = jest
    .spyOn(console, 'error')
    .mockImplementation(() => undefined)

  afterEach(() => {
    consoleSpy.mockReset()
  })

  afterAll(() => {
    consoleSpy.mockRestore()
  })

  it('should be an empty console output for correct style', () => {
    validator()(
      {
        fontSize: '10px',
      },
      RULE_TYPE
    )

    expect(consoleSpy.mock.calls).toHaveLength(0)
  })

  it('should log into the console an error about the incorrect style', () => {
    validator()(
      {
        fontSize: 'abc',
      },
      RULE_TYPE
    )

    expect(consoleSpy.mock.calls).toMatchSnapshot()
  })

  it('should log into the console an error about the incorrect nested style', () => {
    validator()(
      {
        ':hover': {
          fontSize: 'abc',
        },
      },
      RULE_TYPE
    )

    expect(consoleSpy.mock.calls).toMatchSnapshot()
  })

  it('should log into the console an error about the invalid nested property', () => {
    validator()(
      {
        'wrong nested property': {},
      },
      RULE_TYPE
    )

    expect(consoleSpy.mock.calls).toMatchSnapshot()
  })

  it('should be an empty console output for correct keyframe', () => {
    validator()(
      {
        from: { fontSize: '10px' },
        to: { fontSize: '12px' },
      },
      KEYFRAME_TYPE
    )

    expect(consoleSpy.mock.calls).toHaveLength(0)
  })

  it('should log into the console an error about the invalid keyframe', () => {
    validator()(
      {
        from: 10,
        to: 20,
      },
      KEYFRAME_TYPE
    )

    expect(consoleSpy.mock.calls).toMatchSnapshot()
  })

  it('should log into the console an error about the invalid keyframe property', () => {
    validator()(
      {
        start: { fontSize: '10pxp' },
        end: { fontSiz: '12px' },
      },
      KEYFRAME_TYPE
    )

    expect(consoleSpy.mock.calls).toMatchSnapshot()
  })

  it('should log into the console an error about the incorrect style in keyframe', () => {
    validator()(
      {
        from: { fontSize: '10pxp' },
        to: { fontSiz: '12px' },
      },
      KEYFRAME_TYPE
    )

    expect(consoleSpy.mock.calls).toMatchSnapshot()
  })
})
