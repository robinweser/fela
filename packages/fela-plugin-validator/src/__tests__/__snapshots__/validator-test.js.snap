// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator plugin should log into the console an error about the incorrect nested style 1`] = `
Array [
  Array [
    " Invalid property \\"fontSize\\" with value \\"abc\\". Expected (<font-size>) but found 'abc'.",
    Object {
      "property": "fontSize",
      "value": "abc",
    },
  ],
]
`;

exports[`Validator plugin should log into the console an error about the incorrect style 1`] = `
Array [
  Array [
    " Invalid property \\"fontSize\\" with value \\"abc\\". Expected (<font-size>) but found 'abc'.",
    Object {
      "property": "fontSize",
      "value": "abc",
    },
  ],
]
`;

exports[`Validator plugin should log into the console an error about the incorrect style in keyframe 1`] = `
Array [
  Array [
    " Invalid property \\"fontSize\\" with value \\"10pxp\\". Unexpected token '10pxp'.",
    Object {
      "property": "fontSize",
      "value": "10pxp",
    },
  ],
  Array [
    " Invalid property \\"fontSize\\" with value \\"10pxp\\". Rule is empty.",
    Object {
      "property": "fontSize",
      "value": "10pxp",
    },
  ],
  Array [
    " Invalid property \\"fontSiz\\" with value \\"12px\\". Unknown property 'font-siz'.",
    Object {
      "property": "fontSiz",
      "value": "12px",
    },
  ],
]
`;

exports[`Validator plugin should log into the console an error about the invalid keyframe 1`] = `
Array [
  Array [
    " Invalid keyframe value. An object was expected.",
    Object {
      "percentage": "from",
      "style": 10,
    },
  ],
  Array [
    " Invalid keyframe value. An object was expected.",
    Object {
      "percentage": "to",
      "style": 20,
    },
  ],
]
`;

exports[`Validator plugin should log into the console an error about the invalid keyframe property 1`] = `
Array [
  Array [
    " Invalid keyframe property.
        Expected either \`to\`, \`from\` or a percentage value between 0 and 100.",
    Object {
      "percentage": "start",
      "style": Object {
        "fontSize": "10pxp",
      },
    },
  ],
  Array [
    " Invalid keyframe property.
        Expected either \`to\`, \`from\` or a percentage value between 0 and 100.",
    Object {
      "percentage": "end",
      "style": Object {
        "fontSiz": "12px",
      },
    },
  ],
]
`;

exports[`Validator plugin should log into the console an error about the invalid nested property 1`] = `
Array [
  Array [
    " Invalid nested property. Only use nested media queries, pseudo classes, child selectors or &-combinators.
          Maybe you forgot to add a plugin that resolves \\"wrong nested property\\".",
    Object {
      "property": "wrong nested property",
      "value": Object {},
    },
  ],
]
`;
